src/components/map/Map.tsx

```diff
@@ -157,131 +144,264 @@ export const Map = ({
-          <Polyline
-            coordinates={geometryTourData}
-            strokeWidth={2}
-            strokeColor={colors.primary}
-            zIndex={1}
-          />
-        )}
-        {locations?.map((marker, index) => {
+        <MapLibreGL.Camera ref={cameraRef} minZoomLevel={minZoom} />
+
+        {clusters.map((cluster, index) => {
+          const [longitude, latitude] = cluster.geometry.coordinates;
+          const { cluster: isCluster, point_count: pointCount } = cluster.properties;
+
+          if (clusteringEnabled && isCluster) {
+            return renderCluster({
+              clusterColor: colors.primary,
+              geometry: { coordinates: [longitude, latitude] },
+              id: `cluster-${index}`,
+              onPress: () => handleClusterPress(cluster),
+              properties: { point_count: pointCount }
+            });
+          }
+
+          const marker = locations[cluster.properties.index];
           const isActiveMarker = selectedMarker && marker.id === selectedMarker;
           const serviceName = truncateText(marker.serviceName);
           const title = truncateText(marker.title);
 
           return (
-            <Marker
-              centerOffset={marker.iconAnchor || { x: 0, y: -(MARKER_ICON_SIZE / 2) }}
-              coordinate={marker.position}
-              identifier={marker.id}
+            <MapLibreGL.PointAnnotation
+              id={`${index}-${marker.id}`}
+              coordinate={[marker.position.longitude, marker.position.latitude]}
+              onSelected={() => onMarkerPress?.(marker.id)}
               key={`${index}-${marker.id}`}
-              onPress={() => onMarkerPress?.(marker.id)}
-              zIndex={isActiveMarker ? 1010 : 1}
             >
-              {!!marker.iconName &&
-              marker.iconName != 'ownLocation' &&
-              marker.iconName != 'location' ? (
-                <>
-                  <MapIcon
-                    iconColor={
-                      marker.iconBackgroundColor
-                        ? isActiveMarker
-                          ? marker.iconColor
-                          : marker.iconBackgroundColor
-                        : isActiveMarker
-                        ? colors.accent
-                        : undefined
-                    }
-                  />
-                  <View
-                    style={[
-                      styles.mapIconOnLocationMarker,
-                      isActiveMarker ? styles.mapIconOnLocationMarkerActive : undefined,
-                      !!marker.iconBackgroundColor && {
-                        backgroundColor: isActiveMarker
-                          ? marker.iconColor
-                          : marker.iconBackgroundColor
-                      }
-                    ]}
-                  >
+              <View
+                style={[
+                  styles.markerContainer,
+                  isActiveMarker ? styles.activeMarkerContainer : undefined
```

Comment:
having a styles array you can do `isActiveMarker && styles.activeMarkerContainer` without having the explicit else with undefined in this case

Reply:
since we don't even need `activeMarkerContainer` anymore I can remove the check completely.